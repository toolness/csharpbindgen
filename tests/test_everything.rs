use insta::assert_snapshot_matches;
use std::fs;
use std::path::PathBuf;

use csharpbindgen::{Builder, CSAccess};

fn load_example(name: &'static str) -> String {
    let mut path = PathBuf::new();
    path.push("tests");
    path.push("rust");
    path.push(format!("{}.rs", name));
    fs::read_to_string(path).unwrap()
}

#[test]
fn test_main_example() {
    let example_name = "main_example";
    let code = Builder::new("MyDll", load_example(example_name))
        .class_name("MyStuff")
        .ignore(&["ignore_*", "IGNORE_*", "Ignore*"])
        .access("public_func", CSAccess::Public)
        .access("PublicStruct", CSAccess::Public)
        .generate()
        .unwrap();

    assert_snapshot_matches!(example_name, code);
}
