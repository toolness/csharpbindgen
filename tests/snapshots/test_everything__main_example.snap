---
created: "2019-11-24T12:59:31.538548128Z"
creator: insta@0.8.2
source: tests/test_everything.rs
expression: code
---
// This file has been auto-generated, please do not edit it.

using System;
using System.Runtime.InteropServices;

[Serializable]
[StructLayout(LayoutKind.Sequential)]
internal struct MyStruct {
    internal Single foo;
    internal Single bar;

    internal MyStruct(Single foo, Single bar) {
        this.foo = foo;
        this.bar = bar;
    }
}

[Serializable]
[StructLayout(LayoutKind.Sequential)]
public struct PublicStruct {
    public Int32 bop;

    public PublicStruct(Int32 bop) {
        this.bop = bop;
    }
}

internal class MyStuff {
    internal const Byte BOOP = 1;

    internal delegate Byte CallbackFn(Byte v);

    public delegate void PublicCallbackFn(ref MyStruct s);

    [DllImport("MyDll")]
    public static extern void public_func();

    [DllImport("MyDll")]
    internal static extern Byte blarg(Int32 a, ref MyStruct b, IntPtr /* MyOpaqueStruct */ c);

    [DllImport("MyDll")]
    internal static extern Byte primitive_typeck(Byte _v1, SByte _v2, UInt16 _v3, Int16 _v4, UInt32 _v5, Int32 _v6, Int64 _v7, UInt64 _v8, UIntPtr _v9, IntPtr _v10);

    [DllImport("MyDll")]
    internal static extern Byte fn_with_callback(Byte v, CallbackFn cb);

    [DllImport("MyDll")]
    internal static extern Byte fn_with_callback_ptr(Byte v, IntPtr /* CallbackFn */ cb);

}

